architecture/bela.cpp

```diff
@@ -315,6 +334,142 @@ enum EInOutPin
     kNumInputPins
 };
 
+const char* const NoteStrings[] =
+{
+    "A",
+    "A#",
+    "B",
+    "C",
+    "C#",
+    "D",
+    "D#",
+    "E",
+    "F",
+    "F#",
+    "G",
+    "G#"
+};
+
+/**
+ * @brief parseBracketString, parse an arbitrary {} string {...} and store the result in s
+ * @param p the string to parse, then the remaining string
+ * @param quote the character used to quote the string
+ * @param s the string found if any
+ * @return true if a string was found at the begin of p
+ */
+static bool parseBracketString(const char*& p, string& s)
```

Comment:
You are using part of https://github.com/grame-cncm/faust/blob/master-dev/architecture/faust/gui/SimpleParser.h right? Then I suggest including the file, and possibly completing it with new functions, instead of coping the code in bela.cpp

Reply:
no, the file is already in the include directives, i have create 4 new functions (based on the SimplerParser.h code) for parsing the parameters. I hesitated to including the new functions in the SimpleParser.h file but I wanted to take the time to think about their general interest and maybe their names before